src/libUtils/Logger.cpp

```diff
@@ -48,9 +49,13 @@ inline pid_t getCurrentPid() {
 const streampos Logger::MAX_FILE_SIZE =
     1024 * 1024 * 100;  // 100MB per log file
 
-Logger::Logger(const char* prefix, bool log_to_file, streampos max_file_size) {
+Logger::Logger(const char* prefix, bool log_to_file, const char* logpath,
+               streampos max_file_size) {
   this->m_logToFile = log_to_file;
   this->m_maxFileSize = max_file_size;
+  this->m_logPath = logpath;
+
+  boost::filesystem::create_directory(logpath);
```

Comment:
The directory creation could fail so some error handling multiple be needed here.